packages/sdk/client-services/src/packlets/identity/identity-manager.ts

```diff
@@ -204,6 +243,13 @@ export class IdentityManager {
       displayName: this._identity.profileDocument?.displayName,
     });
 
+    if (!this.currentDeviceProfile) {
```

Comment:
can we pass this as a parameter to `acceptIdentity` (and transitively other flows that trigger this)? This should avoid this extra state in the service

Reply:
yeah, I had failed implementing it in that manner by modifying the `DeviceInvitationProtocol` but now see the problem.